test/foundry/ERC4626TrancheWrapper.t.sol

```diff
@@ -0,0 +1,142 @@
+    address public owner;
+    IdleCDO internal idleCDO;
+    IERC20Detailed internal underlying;
+    IERC20Detailed internal strategyToken;
+    IdleCDOTranche internal AAtranche;
+    IdleCDOTranche internal BBtranche;
+    IIdleCDOStrategy internal strategy;
+    bytes internal extraData;
+    bytes internal extraDataSell;
+
+    TrancheWrapper internal trancheWrapper;
+
+    function setUp() public override {
+        setUpIdleCDO();
+
+        // deploy trancheWrapper
+        address tranche = idleCDO.AATranche();
+        trancheWrapper = new TrancheWrapper(idleCDO, tranche);
+
+        _underlying_ = address(underlying);
+        _vault_ = address(trancheWrapper);
+        _delta_ = 10;
+
+        // fund
+        uint256 initialBal = 1e10 * ONE_SCALE;
+        deal(address(underlying), address(this), initialBal, true);
+        deal(address(underlying), address(strategy.strategyToken()), initialBal, true);
+
+        // label
+        vm.label(address(idleCDO), "idleCDO");
+        vm.label(address(AAtranche), "AAtranche");
+        vm.label(address(BBtranche), "BBtranche");
+        vm.label(address(strategy), "strategy");
+        vm.label(address(underlying), "underlying");
+        vm.label(address(strategyToken), "strategyToken");
+    }
+
+    function setUpIdleCDO() public {
+        owner = address(2);
+        address _rebalancer = address(3);
+        address[] memory _incentiveTokens = new address[](0);
+
+        (address _strategy, address _underlying) = _deployStrategy(owner);
+        idleCDO = _deployIdleCDO();
+
+        stdstore.target(address(idleCDO)).sig(idleCDO.token.selector).checked_write(address(0));
+        idleCDO.initialize(
+            0,
+            _underlying,
+            address(this), // governanceFund,
+            owner, // owner,
+            _rebalancer, // rebalancer,
+            _strategy, // strategyToken
+            20000, // apr split: 100000 is 100% to AA
+            50000, // ideal value: 50% AA and 50% BB tranches
+            _incentiveTokens
+        );
+
+        underlying = IERC20Detailed(idleCDO.token());
+        decimals = underlying.decimals();
+        ONE_SCALE = 10**decimals;
+        strategy = IIdleCDOStrategy(idleCDO.strategy());
+        strategyToken = IERC20Detailed(strategy.strategyToken());
+        AAtranche = IdleCDOTranche(idleCDO.AATranche());
+        BBtranche = IdleCDOTranche(idleCDO.BBTranche());
+
+        vm.startPrank(owner);
+        idleCDO.setIsAYSActive(true);
+        idleCDO.setUnlentPerc(0);
+        idleCDO.setFee(0);
+        idleCDO._setLimit(0);
+
+        IdleStrategy(address(strategy)).setWhitelistedCDO(address(idleCDO));
+        vm.stopPrank();
+    }
+
+    function _deployStrategy(address _owner) internal returns (address _strategy, address _underlying) {
+        _underlying = address(new MockERC20("MockDAI", "MockDAI"));
+        MockIdleToken idleToken = new MockIdleToken(_underlying);
+        idleToken.setTokenPriceWithFee(12**17);
```

Comment:
Why are you using a Mock here? In general if we can test we live contract would be better

Reply:
if i tried live contract, it takes much time to run fuzzing tests